src/prefect/runner/server.py

```diff
@@ -49,8 +71,76 @@ def _shutdown():
     return _shutdown
 
 
+def _model_for_function(fn: t.Callable) -> t.Type[BaseModel]:
+    signature = inspect.signature(fn)
+    model_fields = {}
+    docstrings = fn.__doc__ or {}
+    aliases = {}
+
+    class ModelConfig:
+        arbitrary_types_allowed = True
+
+    if HAS_PYDANTIC_V2 and has_v2_model_as_param(signature):
+        create_schema = create_v2_schema
+        process_params = process_v2_params
+    else:
+        create_schema = create_v1_schema
+        process_params = process_v1_params
+
+    for position, param in enumerate(signature.parameters.values()):
+        name, type_, field = process_params(
+            param, position=position, docstrings=docstrings, aliases=aliases
+        )
+        # Generate a Pydantic model at each step so we can check if this parameter
+        # type supports schema generation
+        try:
+            create_schema(
+                "CheckParameter", model_cfg=ModelConfig, **{name: (type_, field)}
+            )
+        except ValueError:
+            # This field's type is not valid for schema creation, update it to `Any`
+            type_ = t.Any
+        model_fields[name] = (type_, field)
+    return create_model(
+        f"{fn.__name__.title()}FunctionModel", __config__=ModelConfig, **model_fields
+    )
+
+
+async def __run_deployment(deployment: "Deployment"):
+    assert deployment.entrypoint is not None
+    _flow = load_flow_from_entrypoint(deployment.entrypoint)
```

Comment:
you should be able to retrieve the parameter openAPI schema on the deployment from the API, I recommend doing that to keep this process lightweight

Reply:
Hey @cicdw, we flip-flopped a bit with the best way to generate the openapi schema on the new endpoint but we ultimately figured out a way to swap it into the FastAPI app directly. All we're doing now is copying the deployment's schema into the auto-generated schema's dicts!